python/ray/tune/automl/search_policy.py

```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/env python
+# Author: Rujie Jiang rujie.jrj@antfin.com
+# Date: Tue May 15 11:46:20 2018
+
+import time
+import copy
+import logging
+
+from ray.tune.trial import Trial
+from ray.tune.suggest import SearchAlgorithm
+from ray.tune.suggest.variant_generator import generate_variants
+from ray.tune.config_parser import make_parser, create_trial_from_spec
+
+
+def merge_config(path, value, config):
+    """Merge kv pair into config by expanding path separated by dot."""
+    keys = path.split('.')
+    for name in keys[:-1]:
+        if config.get(name) is None:
```

Comment:
I think `config.setdefault(name, {})` works fine here instead of this case.

Also, are you sure this works as intended? `config = config[name]` seems odd. Would be great to get some tests for this.

Reply:
I think the following can be cleaner - 

```python

def deep_insert(path_list, value, config):
    """Inserts value into config by path, generating intermediate dictionaries.

    Example:
        >>> deep_insert(path.split("."), value, {})
    """
    if len(path_list) > 1:
        inside_config = config.setdefault(path_list[0], {})
        deep_insert(path_list[1:], value, inside_config)
    else:
       config[path_list[0]] = value

Reply:
Yep, good point
Done